src/Tes/Repository/TesTaskPostgreSqlRepository.cs

```diff
@@ -0,0 +1,169 @@
+
+        /// <summary>
+        /// Get a TesTask by the TesTask.ID
+        /// </summary>
+        /// <param name="id">The TesTask's ID</param>
+        /// <param name="onSuccess">Delegate to be invoked on success</param>
+        /// <returns></returns>
+        public async Task<bool> TryGetItemAsync(string id, Action<TesTask> onSuccess = null)
+        {
+            using var dbContext = createDbContext();
+
+            // Search for Id within the JSON
+            var item = await dbContext.TesTasks.FirstOrDefaultAsync(t => t.Json.Id == id);
+
+            if (item is null)
+            {
+                return false;
+            }
+
+            onSuccess?.Invoke(item.Json);
+            return true;
+        }
+
+        /// <summary>
+        /// Get TesTask items
+        /// </summary>
+        /// <param name="predicate">Predicate to query the TesTasks</param>
+        /// <returns></returns>
+        public async Task<IEnumerable<TesTask>> GetItemsAsync(Expression<Func<TesTask, bool>> predicate)
+        {
+            using var dbContext = createDbContext();
+
+            // Search for items in the JSON
+            var query = dbContext.TesTasks.Select(t => t.Json).Where(predicate);
+            
+            //var sqlQuery = query.ToQueryString();
+            //Debugger.Break();
+            return await query.ToListAsync();
+        }
+
+        /// <summary>
+        /// Encapsulates a TesTask as JSON
+        /// </summary>
+        /// <param name="item">TesTask to store as JSON in the database</param>
+        /// <returns></returns>
+        public async Task<TesTask> CreateItemAsync(TesTask item)
+        {
+            using var dbContext = createDbContext();
+            var dbItem = new TesTaskDatabaseItem { Json = item };
+            dbContext.TesTasks.Add(dbItem);
+            await dbContext.SaveChangesAsync();
+            return item;
+        }
+
+        /// <summary>
+        /// Encapsulates TesTasks as JSON
+        /// </summary>
+        /// <param name="item">TesTask to store as JSON in the database</param>
+        /// <returns></returns>
+        public async Task<List<TesTask>> CreateItemsAsync(List<TesTask> items)
+        {
+            using var dbContext = createDbContext();
+
+            foreach (var item in items) 
+            {
+                var dbItem = new TesTaskDatabaseItem { Json = item };
+                dbContext.TesTasks.Add(dbItem);
+            }
+
+            await dbContext.SaveChangesAsync();
+            return items;
+        }
+
+        /// <summary>
+        /// Base class searches within model, this method searches within the JSON
+        /// </summary>
+        /// <param name="tesTask"></param>
+        /// <returns></returns>
+        public async Task<TesTask> UpdateItemAsync(TesTask tesTask)
+        {
+            using var dbContext = createDbContext();
+            var item = await dbContext.TesTasks.FirstOrDefaultAsync(t => t.Json.Id == tesTask.Id);
```

Comment:
The `select` here (await... FirstOrDefaultAsync()) is completed in a single transaction, and the update in a single transaction.  No TCP connection is held open between the `FirstOrDefaultAsync()` and `SaveChangesAsync()`.  The select is so that EF can track changes and then generate update SQL that only modifies the changed properties.   PostgreSQL takes a row lock during the update.